app/assets/javascripts/discourse/tests/acceptance/sidebar-plugin-api-test.js

```diff
@@ -0,0 +1,255 @@
+                  get name() {
+                    "random-channel";
+                  }
+                  get route() {
+                    return "discovery.latest";
+                  }
+                  get model() {
+                    return false;
+                  }
+                  get title() {
+                    return "random channel title";
+                  }
+                  get text() {
+                    return "random channel text";
+                  }
+                  get prefixIcon() {
+                    return "hashtag";
+                  }
+                  get prefixIconColor() {
+                    return "FF0000";
+                  }
+                  get prefixIconBadge() {
+                    return "lock";
+                  }
+                  get suffixIcon() {
+                    return "circle";
+                  }
+                  get suffixCssClass() {
+                    return "unread";
+                  }
+                })(),
+                new (class extends BaseCustomSidebarSectionLink {
+                  get name() {
+                    "dev-channel";
+                  }
+                  get route() {
+                    return "discovery.latest";
+                  }
+                  get model() {
+                    return false;
+                  }
+                  get title() {
+                    return "dev channel title";
+                  }
+                  get text() {
+                    return "dev channel text";
+                  }
+                  get prefixIconColor() {
+                    return "alert";
+                  }
+                  get prefixIcon() {
+                    return "hashtag";
+                  }
+                })(),
+              ];
+            }
+          };
+        }
+      );
+    });
+
+    await visit("/");
+    assert.strictEqual(
+      query(".sidebar-section-chat-channels .sidebar-section-header a").title,
+      "chat channels title",
+      "displays header with correct title attribute"
+    );
+    assert.strictEqual(
+      query(
+        ".sidebar-section-chat-channels .sidebar-section-header a"
+      ).textContent.trim(),
+      "chat channels text",
+      "displays header with correct text"
+    );
+    await click(".edit-channels-dropdown summary");
+    assert.strictEqual(
+      queryAll(".edit-channels-dropdown .select-kit-collection li").length,
+      2,
+      "displays two actions"
+    );
+    const $actions = queryAll(
+      ".edit-channels-dropdown .select-kit-collection li"
+    );
+    assert.strictEqual(
+      $actions[0].textContent.trim(),
+      "Browse channels",
+      "displays first header action with correct text"
+    );
+    assert.strictEqual(
+      $actions[1].textContent.trim(),
+      "Settings",
+      "displays second header action with correct text"
+    );
+
+    const $links = queryAll(
```

Comment:
```suggestion
    const links = queryAll(
```

Ditto: https://github.com/discourse/discourse/pull/17296/files#r919564731